proposals/src/mappings/mappingHandlers.ts

```diff
@@ -80,26 +84,28 @@ const updateOrCreateAndGetProposal = async (
         "at_height" in expiration ? expiration.at_height : undefined;
     } catch (err) {
       logger.error(
-        `Error retrieving expiration for ${id}: ${
-          err instanceof Error ? err.message : `${err}`
-        }`
+        `Error retrieving expiration for ${getProposalId(
+          proposalModuleAddress,
+          proposalNumber
+        )}: ${err instanceof Error ? err.message : `${err}`}`
       );
       return;
     }
 
     proposal = Proposal.create({
-      id,
+      id: getProposalId(proposalModuleAddress, proposalNumber),
       moduleId: proposalModuleAddress,
       num: proposalNumber,
       open: shouldBeOpen,
       expiresAtDate,
       expiresAtHeight,
+      createdAt: blockDate,
     });
     await proposal.save();
   }
 
   // Update proposal open status if necessary.
-  if (proposal && proposal.open !== shouldBeOpen) {
+  if (proposal.open !== shouldBeOpen) {
     proposal.open = shouldBeOpen;
     await proposal.save();
   }
```

Comment:
```suggestion
  proposal.open = shouldBeOpen;
  await proposal.save();
```

this is logically equivalent to just always assigning and saving. figure its a little more clear to just do that every time if that's whats intended? whatever you think is more readable.

Reply:
i think this avoids unnecessary saves by not saving to the graphql db if open is already correct

Reply:
sweet :) would be interesting to measure if that has a performance impact, but will probably be pretty deep into subquery before that starts to matter lol